src/brogue/Time.c

```diff
@@ -537,14 +537,10 @@ void applyGradualTileEffectsToCreature(creature *monst, short ticks) {
         && !(monst->info.flags & MONST_INANIMATE)
         && !(monst->bookkeepingFlags & MB_SUBMERGED)) {
 
-        damage = (monst->info.maxHP / 15) * ticks / 100;
-        damage = max(1, damage);
-        if (monst->currentHP < monst->info.maxHP) {
-            monst->currentHP = min(monst->currentHP + damage, monst->info.maxHP);
-            if (monst == &player) {
-                messageWithColor("you feel much better.", &goodMessageColor, 0);
-            }
+        if (monst == &player && monst->currentHP < monst->info.maxHP) {
+            messageWithColor("you feel much better.", &goodMessageColor, 0);
         }
+        heal(monst, ticks / 15, true);
```

Comment:
Just an observation: this refactor changes the rounding behaviour:

```
// new heal amount
max(1, (ticks / 15) * monst->info.maxHP / 100)
// old
max(1, (monst->info.maxHP / 15) * ticks / 100)
```

Not sure what effect this has but it's probably minor

Reply:
Yes, I was aware of the rounding behavior.  I also decided it was unlikely to alter the gameplay significantly, and it allowed refactoring the health calculation from two different files into a single place.